### PR TITLE
add support for python 3.12

### PR DESCRIPTION
**Pull request by krmax44**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -52,7 +52,7 @@ jobs:
           DATABASE_URL: postgis://postgres:postgres@localhost/fragdenstaat_de
     services:
       postgres:
-        image: postgis/postgis:14-3.3
+        image: postgis/postgis:16-3.4
```
Comment (pajowu) [inline .github/workflows/ci.yml:55]: This should probably also be reflected in the docker-compose files

File: fragdenstaat_de/settings/base.py [hunk 2]
```diff
@@ -49,7 +49,6 @@ def INSTALLED_APPS(self):
                 "djangocms_transfer",
                 # easy thumbnails comes from froide
                 "filer",
-                "mptt",
```
Comment (pajowu) [inline fragdenstaat_de/settings/base.py:52]: Why is this needed? Are there any implications?

File: fragdenstaat_de/settings/base.py [hunk 3]
```diff
@@ -375,6 +374,7 @@ def three_days_ago_but_not_sundays(date):
     }
 
     FILER_ENABLE_PERMISSIONS = True
+    FILER_MIME_TYPE_WHITELIST = ["*"]
```
Comment (pajowu) [inline fragdenstaat_de/settings/base.py:377]: Why is this needed? Are there any implications?

File: requirements.in [hunk 4]
```diff
@@ -75,7 +75,6 @@ Wand
 WeasyPrint
 webp
 websockets
-xlrd
```
Comment (pajowu) [inline requirements.in:78]: Why? From what I can tell, pandas still needs `xlrd` to read `.xls` files

File: fragdenstaat_de/settings/base.py (hunk 2 above)
Comment (krmax44) [reply fragdenstaat_de/settings/base.py:52]: This used to be a dependency of django-filer, see [upgrading notes](https://django-filer.readthedocs.io/en/latest/upgrading.html). Should have probably documented all these questions beforehand, sorry!

File: fragdenstaat_de/settings/base.py (hunk 3 above)
Comment (krmax44) [reply fragdenstaat_de/settings/base.py:377]: django-filer 3.0 introduces file validation. Previously, all files were accepted without checks. That line keeps this behaviour. We could configure it more strictly, see [docs](https://django-filer.readthedocs.io/en/latest/validation.html#validation).

File: requirements.in (hunk 4 above)
Comment (krmax44) [reply requirements.in:78]: xlrd isn't really maintained anymore, and doesn't support Python 3.12. I could only find us using pandas in some internal, donation related tool, where we probably don't encounter old `.xls` files. Do we rely on pandas' Excel features indirectly in other places?

File: fragdenstaat_de/settings/base.py (hunk 3 above)
Comment (pajowu) [reply fragdenstaat_de/settings/base.py:377]: But isnt this equivalent to the default? The docs say

> If FILER_MIME_TYPE_WHITELIST is empty, all mime types will be accepted (default behaviour).

File: requirements.in (hunk 4 above)
Comment (pajowu) [reply requirements.in:78]: Okay. I think we should be good. Was just wondering why this was in here
